src/main/java/de/unijena/cheminf/fragment/fingerprint/performanceTest/PerformanceTest.java

```diff
@@ -0,0 +1,419 @@
+            List<String> tmpListOfMolecule = (List<String>) tmpMolecule.next();
+            String tmpMoleculeNameOrID = (String) tmpMoleculeNameIterator.next();
+            tmpBitArray = tmpFragmentFingerprinter.getBitArray(tmpListOfMolecule);
+            this.bitArrayPrintWriter.println(tmpMoleculeNameOrID + ":" + java.util.Arrays.toString(tmpBitArray));
+        }
+        this.resultsPrintWriter.println();
+        this.resultsPrintWriter.println("#########################################################################");
+        this.resultsPrintWriter.println("\n\tGenerate count fingerprints");
+        this.resultsPrintWriter.println();
+        this.countPrintWriter.println("Number of fragments: " + this.fragmentList.size() + " and number of molecules: " + this.moleculeListForBitFingerprint.size());
+        this.countPrintWriter.println("Number of molecules in process, Count fingerprint process time in ms.");
+        for(int i = aNumberOfMoleculesInProcess; i<= this.moleculeFragmentList.size(); i+= aNumberOfMoleculesInProcess) {  //i+=2000)
+            List<HashMap<String, Integer>> tmpNumberOfMoleculesInProcessList = this.moleculeFragmentList.subList(0, i);
+            long start = System.currentTimeMillis();
+            for (HashMap<String, Integer> tmpUniqueSmilesToFrequencyMap : tmpNumberOfMoleculesInProcessList) {
+                try {
+                    ICountFingerprint count = tmpFragmentFingerprinter.getCountFingerprint(tmpUniqueSmilesToFrequencyMap);
+                } catch (Exception anException) {
+                    this.exceptionsPrintWriter.println("Count fingerprint generation ERROR. There may be incorrect/invalid elements in the fragment list oder molecule list");
+                    this.appendToLogfile(anException);
+                    throw new Exception("Count fingerprint generation ERROR. There may be incorrect/invalid elements in the fragment list oder molecule list");
+                }
+            }
+            long end = System.currentTimeMillis();
+            this.resultsPrintWriter.println("Processing " + tmpNumberOfMoleculesInProcessList.size() + " valid molecules.");
+            this.resultsPrintWriter.println("Count fingerprint generation took: " + (end - start) + " ms.");
+            this.resultsPrintWriter.println();
+            this.countPrintWriter.println(tmpNumberOfMoleculesInProcessList.size() + "," + (end - start));
+        }
+    }
+    //
+    /**
+     * Appends the given exception's stack trace to a log file.
+     *
+     * @param anException the exception to log
+     */
+    private void appendToLogfile(Exception anException) {
+        if (anException == null) {
+            return;
+        }
+        try {
+            FileWriter tmpFileWriter = new FileWriter(this.workingPath
+                    + "/Results/" + PerformanceTest.EXCEPTIONS_LOG_FILE_NAME, true);
```

Comment:
The file writer is unused

Reply:
...and never closed.
Always close a writer or a reader in a finally block. Or use the try-with-resources statement.